relations/domain.py

```diff
@@ -1,3 +1,43 @@
 """Core domain concepts and rules."""
 
-from typing import NamedTuple
\ No newline at end of file
+from datetime import datetime
+from enum import Enum, auto
+from typing import NamedTuple, Optional
+
+ArXivID = str
+RelationID = str
+
+
+class EPrint(NamedTuple):
+    """Info of E-print."""
+
+    arxiv_id: ArXivID
+    version: int
+
+
+class RelationType(Enum):
+    """Type of relations."""
+
+    ADD = auto()
+    EDIT = auto()
+    DEACTIVATE = auto()
+
+
+class Resource(NamedTuple):
+    """Domain class of resources."""
+
+    resource_type: str
+    identifier: str
+
+
+class Relation(NamedTuple):
+    """The core domain class of relations."""
+
+    identifier: RelationID
+    relation_type: RelationType
+    e_print: EPrint
+    resource: Resource
+    description: str
+    added_at: datetime
+    creator: Optional[str]
+    superceded_or_suppressed: Optional[RelationID]
```

Comment:
Is this a reference to the a relation that is superceded or suppressed by this relation? Maybe put it in present tense, e.g. ``supercedes_or_suppresses``?

Reply:
6d2a39de7dbea994c3f9f515bb57feca3a109ba7